libraries/AP_LandingGear/AP_LandingGear.cpp

```diff
@@ -208,6 +233,30 @@ void AP_LandingGear::update()
         
         gear_state_current = gear_state_new;
     }
+
+    /*
+      check for height based triggering
+     */
+    int alt_m = constrain_int16(height_above_ground_m, 0, 127);
```

Comment:
Do we want "int16_t" here perhaps?

Reply:
I didn't think deploy of landing gear above 127m is really useful. I can change it if you like. I guess it could be a 747 ...

Reply:
I've changed to int16_t